packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -658,13 +628,17 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                 const odspSnapshot = await fetchLatestSnapshotCore(
                     this.odspResolvedUrl,
                     this.getStorageToken,
-                    tokenFetchOptions,
                     snapshotOptionsWithoutBlobs,
                     this.logger,
                     snapshotDownloader,
                     putInCache);
                 return odspSnapshot;
             }
+            // Clear the cache on 401/403/404 on snapshot fetch from network because this means either the user doesn't have
```

Comment:
I think logically this should move to fetchLatestSnapshotCore(), as it is applicable to pre-fetching as well, right?

Reply:
I will handle this in my next PR about redeeming.